apps/frontend/src/libs/components/protected-route/protected-route.tsx

```diff
@@ -0,0 +1,16 @@
+import { Navigate, Outlet } from "react-router-dom";
+
+import { AppRoute, DataStatus } from "~/libs/enums/enums.js";
+import { useAppSelector } from "~/libs/hooks/hooks.js";
+
+const ProtectedRoute: React.FC = () => {
+	const authStatus = useAppSelector(({ auth }) => auth.dataStatus);
+
+	if (authStatus === DataStatus.FULFILLED) {
+		return <Outlet />;
+	}
+
+	return <Navigate replace to={AppRoute.SIGN_IN} />;
```

Comment:
You shouldn't check the status. You should check whether the user is present or not